### PR TITLE
feat: Support the users.verify_password endpoint

### PR DESCRIPTION
**Pull request by agis**

### CODE REVIEW

File: lib/clerk/resources/users.rb [hunk 1]
```diff
@@ -16,6 +16,14 @@ def initialize(client)
       def oauth_access_token(user_id, provider)
         @client.request(:get, "#{@resource.resource_path(user_id)}/oauth_access_tokens/#{provider}")
       end
+
+      def verify_password(user_id, password)
+        res = @client.request(:post, "#{@resource.resource_path(user_id)}/verify_password", body: { password: password })
+
+        return true if res == ""
```
Comment (gkats) [inline lib/clerk/resources/users.rb:23]: :question: What is this line used for?

Comment (agis) [reply lib/clerk/resources/users.rb:23]: The API responds with 204 No Content and I found `true` a more intuitive return value than `""` for this specific method.

Comment (gkats) [reply lib/clerk/resources/users.rb:23]: I see, thanks for explaining. 

I'm slowly starting to think that perhaps "No Content" and the omission of a response body is not a good API design practice. Most of the time API consumers expect to parse a response body. Forcing them to deal with 204 responses in a different way might not be ideal.

In our case, for this client, I'm also questioning whether return either true or a response (which is also a truthy value) is a good design. What do we do with all other responses? Is there also special handling for other methods?

Final question, how would somebody use this client method? I suppose something like the following?

```ruby
res = verify_password

unless res == true
  # do something with the response body, probably error handling
end
```

Comment (SokratisVidros) [reply lib/clerk/resources/users.rb:23]: @gkats I see your point and I agree. This creates a bigger hassle to the developer. @agis Can we revert back to the original decision and return `{verified: true}`?

Comment (agis) [reply lib/clerk/resources/users.rb]: Updated, PTAL.

Comment (agis) [reply lib/clerk/resources/users.rb]: @SokratisVidros @gkats pretty ping

**Pull request merged**